lib/link_preview/parser.rb

```diff
@@ -84,38 +87,74 @@ def parse_html(data)
         discovered_uris << LinkPreview::URI.parse(link_rel, @options)
       end
 
-      opengraph_image_array_first_elem = find_meta_property_array(doc, 'og:image').first
-      opengraph_video_array_first_elem = find_meta_property_array(doc, 'og:video').first
+      {
+        opengraph: parse_opengraph_video_data(doc),
+        opengraph_embed: parse_opengraph_embed_data(doc),
+        html: parse_html_data(doc)
+      }
+    end
 
+    def parse_html_data(doc)
       {
-        opengraph: {
-          title: find_meta_property(doc, 'og:title'),
-          description: find_meta_property(doc, 'og:description'),
-          image_secure_url: opengraph_image_array_first_elem['og:image:secure_url'],
-          image_url: opengraph_image_array_first_elem['og:image'] || opengraph_image_array_first_elem['og:image:url'],
-          tag: find_meta_property(doc, 'og:tag'),
-          url: find_meta_property(doc, 'og:url'),
-          type: find_meta_property(doc, 'og:type'),
-          site_name: find_meta_property(doc, 'og:site_name'),
-          video_secure_url: opengraph_video_array_first_elem['og:video:secure_url'],
-          video_url: opengraph_video_array_first_elem['og:video'] || opengraph_video_array_first_elem['og:video:url'],
-          video_type: opengraph_video_array_first_elem['og:video:type'],
-          video_width: opengraph_video_array_first_elem['og:video:width'],
-          video_height: opengraph_video_array_first_elem['og:video:height']
-        },
-        html: {
-          title: find_title(doc),
-          description: find_meta_description(doc),
-          tags: Array.wrap(find_rel_tags(doc))
-        }
+        title: find_title(doc),
+        description: find_meta_description(doc),
+        tags: Array.wrap(find_rel_tags(doc))
       }
     end
 
+    # FIXME: currently secure_url is favored over url via implicit ordering of keys
```

Comment:
IMO it's much better to file a follow-up Jira ticket than to put todo-type comments into the source.